libs/api/domains/vehicles/src/lib/api-domains-vehicles-mileage.resolver.ts

```diff
@@ -0,0 +1,100 @@
+import {
+  Args,
+  Context,
+  Mutation,
+  Parent,
+  Query,
+  ResolveField,
+  Resolver,
+} from '@nestjs/graphql'
+import { UseGuards } from '@nestjs/common'
+import {
+  IdsUserGuard,
+  ScopesGuard,
+  Scopes,
+  CurrentUser,
+} from '@island.is/auth-nest-tools'
+import type { User } from '@island.is/auth-nest-tools'
+import { ApiScope } from '@island.is/auth/scopes'
+import { Audit } from '@island.is/nest/audit'
+import { VehiclesService } from './api-domains-vehicles.service'
+import {
+  VehicleMileageDetail,
+  VehicleMileageOverview,
+  VehicleMileagePutModel,
+} from '../models/getVehicleMileage.model'
+import { GetVehicleMileageInput } from '../dto/getVehicleMileageInput'
+import {
+  PostVehicleMileageInput,
+  PutVehicleMileageInput,
+} from '../dto/postVehicleMileageInput'
+
+@UseGuards(IdsUserGuard, ScopesGuard)
+@Resolver(() => VehicleMileageOverview)
+@Audit({ namespace: '@island.is/api/vehicles' })
+export class VehiclesMileageResolver {
+  constructor(private readonly vehiclesService: VehiclesService) {}
+
+  @Scopes(ApiScope.vehicles)
+  @Query(() => VehicleMileageOverview, {
+    name: 'vehicleMileageDetails',
+    nullable: true,
+  })
+  @Audit()
+  async getVehicleMileage(
+    @Args('input') input: GetVehicleMileageInput,
+    @CurrentUser() user: User,
+  ) {
+    return await this.vehiclesService.getVehicleMileage(user, input)
+  }
```

Comment:
~ FYI async/await is unnecessary when returnin the promise from the service function. `return await` is only required inside a `try/catch` when you want to await the response to catch an error.
```suggestion
  getVehicleMileage(
    @Args('input') input: GetVehicleMileageInput,
    @CurrentUser() user: User,
  ) {
    return this.vehiclesService.getVehicleMileage(user, input)
  }
```